python/lsst/meas/extensions/scarlet/multiBandDetectionTask.py

```diff
@@ -0,0 +1,268 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+from typing import Sequence
+
+import lsst.afw.image as afwImage
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+import lsst.scarlet.lite as scl
+import numpy as np
+from lsst.afw.detection import Footprint as AfwFootprint
+from lsst.afw.detection import PeakCatalog
+from lsst.afw.detection.multiband import getSpanSetFromImages
+from lsst.afw.table import IdFactory, Schema, SourceCatalog, SourceTable
+from lsst.geom import Box2I, Point2I
+from lsst.meas.base import SkyMapIdGeneratorConfig
+from scipy.signal import convolve as scipy_convolve
+
+from . import utils
+
+log = logging.getLogger(__name__)
+
+__all__ = [
+    "MultiBandDetectionTask",
+    "MultiBandDetectionConfig",
+    "MultiBandDetectionConnections",
+]
+
+
+class MultiBandDetectionConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "patch", "skymap"),
+    defaultTemplates={"inputCoaddName": "deep"},
+):
+    """Connections for MultiBandDetectionTask"""
+
+    coadds = cT.Input(
+        doc="Exposure on which to run deblending",
+        name="deconvolved_{inputCoaddName}_coadd",
+        storageClass="ExposureF",
+        multiple=True,
+        dimensions=("tract", "patch", "band", "skymap"),
+    )
+
+    peaks = cT.Input(
+        doc="Catalog of detected peak positions",
+        name="{inputCoaddName}_coadd_multiband_peaks",
+        storageClass="PeakCatalog",
+        dimensions=("tract", "patch", "skymap"),
+    )
+
+    footprints = cT.Output(
+        doc="Output catalog of detected footprints",
+        name="deconvolved_{inputCoaddName}_coadd_footprints",
+        storageClass="FootprintCatalog",
+    )
+
+
+class MultiBandDetectionConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=MultiBandDetectionConnections,
+):
+    """Configuration for MultiBandDetectionTask"""
+
+    doDetectPeaks = pexConfig.Field(
+        dtype=bool,
+        doc="Detect peaks in the input images. "
+        "If False then the multi-band peak catalog is used.",
+        default=False,
+    )
+    minFootprintArea = pexConfig.Field(
+        dtype=int,
+        default=4,
+        doc="Minimum area of a footprint to be considered detectable",
```

Comment:
I don't think we typically go so far as to check that counts and areas are positive do we? That seems like overkill for a user error that should be easy to catch.

Reply:
I suppose it depends on whether you want users to be allowed to set it to zero. Would it actually work as expected then?

Reply:
Yes, for a number of reasons. The main one is that the algorithm is incapable of create a zero size footprint (because the pixel being checked will always have itself in the footprint).